app/client/actions/groupMessagesAction.js

```diff
@@ -0,0 +1,78 @@
+import axios from 'axios';
+import {
+  SET_GROUP_MESSAGES,
+  FETCHING_GROUP_MESSAGES,
+  FETCH_GROUP_MESSAGES_FAILURE } from '../actions/types';
+
+
+export function getGroupMessagesForCount(groupId) {
+  const request = axios.get(`/api/group/${groupId}/messages`); // Returns a promise
+
+  return (dispatch) => {
```

Comment:
You are suppose to be dispatching **ACTIONS** not promises, refer to Redux documentation about Actions and Action generators http://redux.js.org/docs/basics/Actions.html, I have noticed you have done that a lot so please review that.